src/cpp/xmlparser/XMLParser.cpp

```diff
@@ -881,15 +913,176 @@ p_dynamictypebuilder_t XMLParser::parseXMLMemberDynamicType(tinyxml2::XMLElement
         return nullptr;
     }
 
+    if (memberType == nullptr)
+    {
+        logError(XMLPARSER, "Error parsing member type: Not found.");
+        return nullptr;
+    }
+
     if (memberArray != nullptr)
     {
         isArray = true;
     }
 
+    if (strcmp(memberType, NON_BASIC_TYPE) == 0)
+    {
+        if (memberNonBasicTypeName != nullptr)
+        {
+            memberType = memberNonBasicTypeName;
+        }
+        else
+        {
+            logError(XMLPARSER, "Error parsing member type: Not found.");
+            return nullptr;
+        }
+    }
+
     types::DynamicTypeBuilder* memberBuilder = nullptr;
     types::DynamicTypeBuilderFactory* factory = types::DynamicTypeBuilderFactory::GetInstance();
 
-    if (strncmp(memberType, BOOLEAN, 8) == 0)
+    if (memberSequence != nullptr)
+    {
+        /*
+            In sequences allowed formats are (complex format includes the basic):
+            sequence<sequence<long,2>,2>
+            <sequence name="my_sequence" length="2">
+                <sequence name="inner_sequence" type="long" length="2"/>
+            </sequence>
+            In this example, inner sequence's name is ignored and can be omited.
+         */
+
+        p_dynamictypebuilder_t contentType;
+        /* FIXME complex sequence
+         if (seqType == nullptr)
+        {
+            contentType = parseXMLMemberDynamicType(p_root->FirstChildElement(), nullptr, MEMBER_ID_INVALID);
+        }
+        else*/
+        {
+
+            contentType = getDiscriminatorTypeBuilder(memberType);
+        }
+
+        if (contentType == nullptr)
+        {
+            logError(XMLPARSER, "Error parsing sequence element type: Cannot be recognized.");
+            return nullptr;
+        }
+
+        const char* lengthStr = p_root->Attribute(SEQ_MAXLENGTH);
+        uint32_t length = MAX_ELEMENTS_COUNT;
+        if (lengthStr != nullptr)
+        {
+            length = std::stoi(lengthStr);
```

Comment:
Be careful, `lengthStr` could include expressions.

Reply:
For future features